module/zfs/spa.c

```diff
@@ -150,6 +151,7 @@ const zio_taskq_info_t zio_taskqs[ZIO_TYPES][ZIO_TASKQ_TYPES] = {
 	{ ZTI_P(12, 8),	ZTI_NULL,	ZTI_ONE,	ZTI_NULL }, /* FREE */
 	{ ZTI_ONE,	ZTI_NULL,	ZTI_ONE,	ZTI_NULL }, /* CLAIM */
 	{ ZTI_ONE,	ZTI_NULL,	ZTI_ONE,	ZTI_NULL }, /* IOCTL */
+	{ ZTI_N(8),	ZTI_NULL,	ZTI_ONE,	ZTI_NULL }, /* TRIM */
```

Comment:
Not that I think it's wrong, but I'm curious about the rationale for `ZTI_N(8)`.  Possibly related, I've been looking into high CPU usage in spin locks due to mutex contention when heavy autotrimming is in progress.

Reply:
I selected 8 initially in order to allow for a synchronous discard operation to be outstanding for multiple leaf vdevs at the same time.  This value worked well in my testing and I expected not to observe contention issues since we also use 8 as in `ZTI_P` above.  That said, I'd definitely rather be conservative here so I'm going to decrease it to 4.